Created Classes

XedinUnknown opened the pull request.

XedinUnknown committed:
Bumped up PHP to 5.4

XedinUnknown committed:
Added dependencies

XedinUnknown committed:
NetBeans project PHP target switched to 5.4

XedinUnknown committed:
Added classes

mecha commented on composer.json:
```diff
@@ -11,13 +11,21 @@
+        "dhii/modular-interface": "dev-task/initial-interface"
+    },
+    "suggest": {
+        "dhii/factory-interface": "For delegating instantiation to a generic factory"
```
Nice touch! :)

mecha commented on src/AbstractFactory.php:
```diff
@@ -0,0 +1,33 @@
+     *
+     * Invoke this in actual constructor.
+     *
+     * @@since [*next-version*]
```
Double `@`

mecha commented on src/AbstractFactory.php:
```diff
@@ -0,0 +1,33 @@
+     *
+     * @@since [*next-version*]
+     */
+    protected function __construct()
```
This has two underscores. Is this intentional?

XedinUnknown committed:
Corrected protected constructor name

XedinUnknown committed:
Corrected `@since` tag

mecha commented:
@XedinUnknown I approve of this after the latest fixes. Are you planning to add tests?

mecha commented on src/AbstractDelegatingFactory.php:
```diff
@@ -0,0 +1,121 @@
+     *
+     * @return string The service ID of the module service.
+     */
+    abstract protected function _getModuleServiceId($config = array());
```
Undocumented `$config` parameter.

mecha commented on src/AbstractDelegatingFactory.php:
```diff
@@ -0,0 +1,121 @@
+     *
+     * @return ModuleFactoryExceptionInterface The new exception.
+     */
+    abstract protected function _createModuleFactoryException($message, ModuleFactoryInterface $factory = null, Exception $inner = null);
```
All parameters are undocumented.

mecha commented on src/AbstractDelegatingFactory.php:
```diff
@@ -0,0 +1,121 @@
+     *
+     * @return CouldNotMakeModuleExceptionInterface The new exception.
+     */
+    abstract protected function _createCouldNotMakeModuleException($message, ModuleFactoryInterface $factory = null, $moduleKey = null, $moduleConfig = null, Exception $inner = null);
```
The `$moduleKey` and `$moduleConfig` parameters are undocumented.

mecha commented on src/AbstractFactory.php:
```diff
@@ -0,0 +1,33 @@
+     *
+     * @return ModuleInterface the new module
+     */
+    abstract protected function _makeModule();
```
Missing `$config` parameter. Is this intentional?

FYI this results in:

```
Fatal error: Declaration of AbstractDelegatingFactory::_makeModule() must be compatible with AbstractFactory::_makeModule()
```

XedinUnknown committed:
Added test

XedinUnknown committed:
Added missing parameter to `_makeModule()`

XedinUnknown committed:
Corrected comment type

XedinUnknown committed:
Removed redundant argument from exception factory methods

This is also because this class does not implement `ModuleInterface`,
and therefore cannot pass itself. However, the factory methods
implemented in the concrete descendants will be able to pass themselves
to the exception constructors, if needed.

XedinUnknown committed:
Corrected exception factory documentation

mecha commented:
@XedinUnknown [This one](https://github.com/RebelCodecom/module-factory-abstract/pull/1/files#r123463098) remained unaddressed.

XedinUnknown committed:
Added param documentation

mecha reviewed (approved):
LGTM

XedinUnknown commented:
@mecha, merged. Please depend on `develop`.
